test/xbgpu/test_engine.py

```diff
@@ -268,6 +268,144 @@ def valid_end_to_end_combination(combo: dict) -> bool:
     return n_ants > 1 and missing_antenna < n_ants
 
 
+def verify_corrprod_sensors(
+    xpipelines: list[XPipeline],
+    corrprod_results: list[np.ndarray],
+    missing_antenna: int | None,
+    incomplete_accumulation_counters: list[int],
+    prom_diff: PromDiff,
+    n_channels_per_substream: int,
+    n_baselines: int,
+    actual_sensor_updates: dict[str, list[tuple[bool | float | str, aiokatcp.Sensor.Status]]],
```

Comment:
This type hint doesn't sit right with me. I know why I needed to do it
* I reused the existing sensor updates list, and
* Reused the sensor observer/callback for the XPipeline's sync-sensor

Couldn't think of anything to resolve my unease; always open to suggestions.